services/echo_console/Console.cpp

```diff
@@ -1002,112 +1011,115 @@ namespace application
+                    throw ConsoleExceptions::IncorrectType{ valueIndex, "integer" };
 
                 formatter.PutVarIntField(value.Get<int64_t>(), field.number);
             }
 
             void VisitInt32(const EchoFieldInt32& field) override
             {
                 if (!value.Is<int64_t>())
-                    throw ConsoleExceptions::IncorrectType{ valueIndex };
+                    throw ConsoleExceptions::IncorrectType{ valueIndex, "integer" };
 
                 formatter.PutVarIntField(value.Get<int64_t>(), field.number);
             }
 
             void VisitFixed32(const EchoFieldFixed32& field) override
             {
                 if (!value.Is<int64_t>())
-                    throw ConsoleExceptions::IncorrectType{ valueIndex };
+                    throw ConsoleExceptions::IncorrectType{ valueIndex, "integer" };
 
                 formatter.PutFixed32Field(static_cast<uint32_t>(value.Get<int64_t>()), field.number);
             }
 
             void VisitFixed64(const EchoFieldFixed64& field) override
             {
                 if (!value.Is<int64_t>())
-                    throw ConsoleExceptions::IncorrectType{ valueIndex };
+                    throw ConsoleExceptions::IncorrectType{ valueIndex, "integer" };
 
                 formatter.PutFixed64Field(static_cast<uint64_t>(value.Get<int64_t>()), field.number);
             }
 
             void VisitBool(const EchoFieldBool& field) override
             {
                 if (!value.Is<bool>())
-                    throw ConsoleExceptions::IncorrectType{ valueIndex };
+                    throw ConsoleExceptions::IncorrectType{ valueIndex, "bool" };
 
                 formatter.PutVarIntField(value.Get<bool>(), field.number);
             }
 
             void VisitString(const EchoFieldString& field) override
             {
                 if (!value.Is<std::string>())
-                    throw ConsoleExceptions::IncorrectType{ valueIndex };
+                    throw ConsoleExceptions::IncorrectType{ valueIndex, "string" };
 
                 formatter.PutStringField(infra::BoundedConstString(value.Get<std::string>().data(), value.Get<std::string>().size()), field.number);
             }
 
             void VisitUnboundedString(const EchoFieldUnboundedString& field) override
             {
                 if (!value.Is<std::string>())
-                    throw ConsoleExceptions::IncorrectType{ valueIndex };
+                    throw ConsoleExceptions::IncorrectType{ valueIndex, "string" };
 
                 formatter.PutStringField(infra::BoundedConstString(value.Get<std::string>().data(), value.Get<std::string>().size()), field.number);
             }
 
             void VisitEnum(const EchoFieldEnum& field) override
             {
                 if (!value.Is<int64_t>())
-                    throw ConsoleExceptions::IncorrectType{ valueIndex };
+                    throw ConsoleExceptions::IncorrectType{ valueIndex, "integer" };
```

Comment:
Possibly create an explicit ConsoleExceptions::IncorrectIntegerType exception? (and string/bool/etc...)

Reply:
I think there are too many different IncorrectTypes; especially were the type is field.protoType